### ISSUE TITLE
"uninitialized constant ActiveRecord" error with mangodb

### ISSUE THREAD

**Issue by lehug** [activerecord-hackery/ransack#718]
my environment is : rails:4.2.6, mongoid : 5.0 , when I put a element in view use "<%= sort_link(@q,:brand) %>",or <%= sort_link(@q,:brand,'Product') %>,it will raise "uninitialized constant ActiveRecord",but when I change gem to "gem 'ransack', github: 'Zhomart/ransack', branch: 'mongoid'",it works.please help me to resolve this question.

errors below:

NameError - uninitialized constant ActiveRecord:
  activesupport (4.2.5) lib/active_support/dependencies.rb:533:in `load_missing_constant'
  activesupport (4.2.5) lib/active_support/dependencies.rb:184:in `const_missing'
  ransack (1.8.2) lib/ransack/helpers/form_helper.rb:144:in `parameters_hash'
  ransack (1.8.2) lib/ransack/helpers/form_helper.rb:101:in `initialize'
  ransack (1.8.2) lib/ransack/helpers/form_helper.rb:53:in `sort_link'
  app/views/fronts/products/search.html.erb:29:in `_app_views_fronts_products_search_html_erb__4058179471897960706_70320948038980'
  actionview (4.2.5) lib/action_view/template.rb:145:in `block in render'
  activesupport (4.2.5) lib/active_support/notifications.rb:164:in `block in instrument'
  activesupport (4.2.5) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  activesupport (4.2.5) lib/active_support/notifications.rb:164:in `instrument'
  actionview (4.2.5) lib/action_view/template.rb:333:in `instrument'
  actionview (4.2.5) lib/action_view/template.rb:143:in `render'
  rack-mini-profiler (0.10.1) lib/mini_profiler/profiling_methods.rb:102:in `block in profile_method'
  actionview (4.2.5) lib/action_view/renderer/template_renderer.rb:54:in `block (2 levels) in render_template'
  actionview (4.2.5) lib/action_view/renderer/abstract_renderer.rb:39:in `block in instrument'
  activesupport (4.2.5) lib/active_support/notifications.rb:164:in `block in instrument'
  activesupport (4.2.5) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  activesupport (4.2.5) lib/active_support/notifications.rb:164:in `instrument'
  actionview (4.2.5) lib/action_view/renderer/abstract_renderer.rb:39:in `instrument'
  actionview (4.2.5) lib/action_view/renderer/template_renderer.rb:53:in `block in render_template'
  actionview (4.2.5) lib/action_view/renderer/template_renderer.rb:61:in `render_with_layout'
  actionview (4.2.5) lib/action_view/renderer/template_renderer.rb:52:in `render_template'
  actionview (4.2.5) lib/action_view/renderer/template_renderer.rb:14:in `render'
  actionview (4.2.5) lib/action_view/renderer/renderer.rb:42:in `render_template'
  actionview (4.2.5) lib/action_view/renderer/renderer.rb:23:in `render'
  actionview (4.2.5) lib/action_view/rendering.rb:100:in `_render_template'
  actionpack (4.2.5) lib/action_controller/metal/streaming.rb:217:in `_render_template'
  actionview (4.2.5) lib/action_view/rendering.rb:83:in `render_to_body'
  actionpack (4.2.5) lib/action_controller/metal/rendering.rb:32:in `render_to_body'
  actionpack (4.2.5) lib/action_controller/metal/renderers.rb:37:in `render_to_body'
  actionpack (4.2.5) lib/abstract_controller/rendering.rb:25:in `render'
  actionpack (4.2.5) lib/action_controller/metal/rendering.rb:16:in `render'
  actionpack (4.2.5) lib/action_controller/metal/instrumentation.rb:44:in `block (2 levels) in render'
  activesupport (4.2.5) lib/active_support/core_ext/benchmark.rb:12:in `block in ms'
  /Users/jyd-9062/.rvm/rubies/ruby-2.3.1/lib/ruby/2.3.0/benchmark.rb:308:in `realtime'
  activesupport (4.2.5) lib/active_support/core_ext/benchmark.rb:12:in `ms'
  actionpack (4.2.5) lib/action_controller/metal/instrumentation.rb:44:in `block in render'
  actionpack (4.2.5) lib/action_controller/metal/instrumentation.rb:87:in `cleanup_view_runtime'
  sunspot_rails (2.2.5) lib/sunspot/rails/railties/controller_runtime.rb:15:in `cleanup_view_runtime'
  actionpack (4.2.5) lib/action_controller/metal/instrumentation.rb:43:in `render'
  app/controllers/fronts/products_controller.rb:34:in `search'
  actionpack (4.2.5) lib/action_controller/metal/implicit_render.rb:4:in `send_action'
  actionpack (4.2.5) lib/abstract_controller/base.rb:198:in `process_action'
  actionpack (4.2.5) lib/action_controller/metal/rendering.rb:10:in `process_action'
  actionpack (4.2.5) lib/abstract_controller/callbacks.rb:20:in `block in process_action'
  activesupport (4.2.5) lib/active_support/callbacks.rb:117:in `call'
  activesupport (4.2.5) lib/active_support/callbacks.rb:555:in `block (2 levels) in compile'
  activesupport (4.2.5) lib/active_support/callbacks.rb:505:in `call'
  activesupport (4.2.5) lib/active_support/callbacks.rb:92:in `__run_callbacks__'
  activesupport (4.2.5) lib/active_support/callbacks.rb:778:in `_run_process_action_callbacks'
  activesupport (4.2.5) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.5) lib/abstract_controller/callbacks.rb:19:in `process_action'
  actionpack (4.2.5) lib/action_controller/metal/rescue.rb:29:in `process_action'
  actionpack (4.2.5) lib/action_controller/metal/instrumentation.rb:32:in `block in process_action'
  activesupport (4.2.5) lib/active_support/notifications.rb:164:in `block in instrument'
  activesupport (4.2.5) lib/active_support/notifications/instrumenter.rb:20:in `instrument'
  activesupport (4.2.5) lib/active_support/notifications.rb:164:in `instrument'
  actionpack (4.2.5) lib/action_controller/metal/instrumentation.rb:30:in `process_action'
  actionpack (4.2.5) lib/action_controller/metal/params_wrapper.rb:250:in `process_action'
  actionpack (4.2.5) lib/abstract_controller/base.rb:137:in `process'
  actionview (4.2.5) lib/action_view/rendering.rb:30:in `process'
  rack-mini-profiler (0.10.1) lib/mini_profiler/profiling_methods.rb:102:in `block in profile_method'
  actionpack (4.2.5) lib/action_controller/metal.rb:196:in `dispatch'
  actionpack (4.2.5) lib/action_controller/metal/rack_delegation.rb:13:in `dispatch'
  actionpack (4.2.5) lib/action_controller/metal.rb:237:in `block in action'
  actionpack (4.2.5) lib/action_dispatch/routing/route_set.rb:76:in `dispatch'
  actionpack (4.2.5) lib/action_dispatch/routing/route_set.rb:45:in `serve'
  actionpack (4.2.5) lib/action_dispatch/journey/router.rb:43:in `block in serve'
  actionpack (4.2.5) lib/action_dispatch/journey/router.rb:30:in `serve'
  actionpack (4.2.5) lib/action_dispatch/routing/route_set.rb:817:in `call'
  meta_request (0.4.0) lib/meta_request/middlewares/app_request_handler.rb:13:in `call'
  meta_request (0.4.0) lib/meta_request/middlewares/meta_request_handler.rb:13:in `call'
  oneapm_rpm (1.3.7) lib/one_apm/rack/middleware_hooks.rb:14:in `traced_call'
  oneapm_rpm (1.3.7) lib/one_apm/rack/middleware_tracing.rb:67:in `call'
  oneapm_rpm (1.3.7) lib/one_apm/rack/browser_monitoring.rb:21:in `traced_call'
  oneapm_rpm (1.3.7) lib/one_apm/rack/middleware_tracing.rb:67:in `call'
  warden (1.2.6) lib/warden/manager.rb:35:in `block in call'
  warden (1.2.6) lib/warden/manager.rb:34:in `call'
  rack (1.6.4) lib/rack/etag.rb:24:in `call'
  rack (1.6.4) lib/rack/conditionalget.rb:25:in `call'
  rack (1.6.4) lib/rack/head.rb:13:in `call'
  actionpack (4.2.5) lib/action_dispatch/middleware/params_parser.rb:27:in `call'
  actionpack (4.2.5) lib/action_dispatch/middleware/flash.rb:260:in `call'
  message_bus (2.0.2) lib/message_bus/rack/middleware.rb:62:in `call'
  rack (1.6.4) lib/rack/session/abstract/id.rb:225:in `context'
  rack (1.6.4) lib/rack/session/abstract/id.rb:220:in `call'
  actionpack (4.2.5) lib/action_dispatch/middleware/cookies.rb:560:in `call'
  actionpack (4.2.5) lib/action_dispatch/middleware/callbacks.rb:29:in `block in call'
  activesupport (4.2.5) lib/active_support/callbacks.rb:88:in `__run_callbacks__'
  activesupport (4.2.5) lib/active_support/callbacks.rb:778:in `_run_call_callbacks'
  activesupport (4.2.5) lib/active_support/callbacks.rb:81:in `run_callbacks'
  actionpack (4.2.5) lib/action_dispatch/middleware/callbacks.rb:27:in `call'
  actionpack (4.2.5) lib/action_dispatch/middleware/reloader.rb:73:in `call'
  actionpack (4.2.5) lib/action_dispatch/middleware/remote_ip.rb:78:in `call'
  better_errors (2.1.1) lib/better_errors/middleware.rb:84:in `protected_app_call'
  better_errors (2.1.1) lib/better_errors/middleware.rb:79:in `better_errors_call'
  better_errors (2.1.1) lib/better_errors/middleware.rb:57:in `call'
  actionpack (4.2.5) lib/action_dispatch/middleware/debug_exceptions.rb:17:in `call'
  rack-contrib (1.4.0) lib/rack/contrib/response_headers.rb:17:in `call'
  meta_request (0.4.0) lib/meta_request/middlewares/headers.rb:16:in `call'
  web-console (2.3.0) lib/web_console/middleware.rb:28:in `block in call'
  web-console (2.3.0) lib/web_console/middleware.rb:18:in `call'
  actionpack (4.2.5) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'
  railties (4.2.5) lib/rails/rack/logger.rb:38:in `call_app'
  railties (4.2.5) lib/rails/rack/logger.rb:20:in `block in call'
  activesupport (4.2.5) lib/active_support/tagged_logging.rb:68:in `block in tagged'
  activesupport (4.2.5) lib/active_support/tagged_logging.rb:26:in `tagged'
  activesupport (4.2.5) lib/active_support/tagged_logging.rb:68:in `tagged'
  railties (4.2.5) lib/rails/rack/logger.rb:20:in `call'
  quiet_assets (1.1.0) lib/quiet_assets.rb:27:in `call_with_quiet_assets'
  request_store (1.3.1) lib/request_store/middleware.rb:9:in `call'
  actionpack (4.2.5) lib/action_dispatch/middleware/request_id.rb:21:in `call'
  rack (1.6.4) lib/rack/methodoverride.rb:22:in `call'
  rack-contrib (1.4.0) lib/rack/contrib/runtime.rb:18:in `call'
  rack (1.6.4) lib/rack/lock.rb:17:in `call'
  actionpack (4.2.5) lib/action_dispatch/middleware/static.rb:116:in `call'
  rack-contrib (1.4.0) lib/rack/contrib/sendfile.rb:103:in `call'
  rack-mini-profiler (0.10.1) lib/mini_profiler/profiler.rb:278:in `call'
  railties (4.2.5) lib/rails/engine.rb:518:in `call'
  railties (4.2.5) lib/rails/application.rb:165:in `call'
  oneapm_rpm (1.3.7) lib/one_apm/rack/middleware_tracing.rb:67:in `call'
  rack (1.6.4) lib/rack/content_length.rb:15:in `call'
  puma (3.6.0) lib/puma/configuration.rb:225:in `call'
  puma (3.6.0) lib/puma/server.rb:578:in `handle_request'
  puma (3.6.0) lib/puma/server.rb:415:in `process_client'
  puma (3.6.0) lib/puma/server.rb:275:in `block in run'
  puma (3.6.0) lib/puma/thread_pool.rb:116:in `block in spawn_thread'.

Thanks!

**Comment by jonatack** [comment]
I believe this has been fixed. Thanks!

**Comment by mbrinkme** [comment]
In which Version is that issue fixed? I use  ```gem 'ransack',             github: 'activerecord-hackery/ransack'``` and have the same problem ```uninitialized constant ActiveRecord``` if I want to use ```sort_link``` with mongoid (5.1.4).

**Comment by jonatack** [comment]
Hi @mbrinkme, I apologize if there was a misunderstanding in closing this issue -- I may have thought this issue was a different one with a similar error message that was resolved. 

I don't maintain gems in the Ruby ecosystem anymore, including Ransack. But for info, as mentioned elsewhere occasionally in the Mongoid issues, all things pertaining to the Mongoid adapter were added -- and to be maintained by -- the community, since neither myself, Ernie nor Radar used MongoDB. We handled the Active Record side. Cheers :)
